cmd/kn/main.go

```diff
@@ -34,10 +34,12 @@ func main() {
 	defer cleanup()
 	rand.Seed(time.Now().UnixNano())
 	err = core.NewDefaultKnCommand().Execute()
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+	if err.Error() == "subcommand is required" {
 		os.Exit(1)
```

Comment:
Sorry - didn't have time to try this myself (stuck in calls), but if we immediately exit w/o printing anything, what does the user see?

Reply:
I just want to make sure we're not exiting w/o some indication of what went wrong so the user can fix it.

Reply:
So, in the most recent push, the only `Error()` ignored is `subcommand is required`, which is occurring when you run `kn` by itself. So running `kn` will result in the following:

```
Manage your Knative building blocks:

* Serving: Manage your services and release new software to them.
* Eventing: Manage event subscriptions and channels. Connect up event sources.

Usage:
  kn [command]

Available Commands:
  completion  Output shell completion code
  help        Help about any command
  plugin      Plugin command group
  revision    Revision command group
  route       Route command group
  service     Service command group
  source      Event source command group
  trigger     Trigger command group
  version     Prints the client version

Flags:
      --config string        kn config file (default is $HOME/.kn/config.yaml)
  -h, --help                 help for kn
      --kubeconfig string    kubectl config file (default is $HOME/.kube/config)
      --log-http             log http traffic
      --lookup-plugins       look for kn plugins in $PATH
      --plugins-dir string   kn plugins directory (default "~/.kn/plugins")

Use "kn [command] --help" for more information about a command.
```

If the error is not `subcommand is required`, this approach should still print the error as originally implemented.

Reply:
ok thanks.